### DISCUSSION TITLE
unable to open database file

### DISCUSSION THREAD

**Discussion by tjing11** [blakeblackshear/frigate#4222] [General]
[2022-10-31 16:20:48] frigate.app                    INFO    : Starting Frigate (0.12.0-)
unable to open database file

After python3 -m frigate, this error pop up. Do I need to create it on my own or where can I get it?

**Comment by NickM-27** [comment]
Are you trying to run via CLI for development purposes? We don't support running via CLI for normal production use cases.

**Reply by tjing11** [reply]
Yes. I am running it for development. Not production but trying to build up some features around it for internal usage. What I am trying to do is that using VS code and docker to run the source code. After Make and config, I issued python3 -m frigate in VS code CLI. This is where I get this error.
Are you saying I can only install the application not develop it?

**Reply by NickM-27** [reply]
> Are you saying I can only install the application not develop it?

No, it is just not  uncommon that a user wants to run frigate without using docker so I wanted to be clear that is not supported. This is not what you are doing.

Do you have a `debug` folder created inside the main directory? You need to make that otherwise it won't be able to get the path to create the database file.

**Reply by tjing11** [reply]
> Do you have a `debug` folder created inside the main directory? You need to make that otherwise it won't be able to get the path to create the database file.

I see. And I don't have the debug folder created yet. So by what you mean, this is what I understand: create a folder called "debug" in the "frigate-dev" folder which contains the "frigate" folder, and so on. Then I will run the command "make"  in VS code CLI to get the database file path. Am I right?

**Reply by NickM-27** [reply]
Not quite. You should be running `make` from your hosts terminal **BEFORE** running the dev container. You'll also want to create the debug folder before running the dev container.

**Reply by tjing11** [reply]
Great... 
How do I reverse it now? Can I do a "docker-compose stop", then create the folder, then **make**?

**Reply by NickM-27** [reply]
I think docker container -D frigate_dev

Might be slightly different but basically want to delete the frigate_dev container then run those then open in vscode again. 

Could also just run make and add the folder then use vscode command palette to rebuild container.

**Reply by tjing11** [reply]
Sorry for the late reply. I have done what you suggested. The frigate server and Web interface are running. Thanks for the help.
However, I'm having a problem with the test .mp4 video.  I put the .mp4 video in the path that applies in config.yml. 
cameras:
  test:
    ffmpeg:
      inputs:
        - path: /home/.../frigate-dev/debug/clips/car-stopping.mp4
          input_args: -re -stream_loop -1 -fflags +genpts
          roles:
            - detect
            - rtmp
    detect:
      height: 1080
      width: 1920
      fps: 5

When I try to view it on the web interface under the "camera/test" category, the video screen shows an all-green color. And the following error message:
ffmpeg.test.detect             ERROR   : /home/.../frigate-dev/debug/clips/car-stopping.mp4: No such file or directory

Is there a file containing the configuration of the file permission or path indicator, such as nginx.conf? Thanks.

**Reply by NickM-27** [reply]
You're putting the path of the file on your host but of course frigate is running in docker and can not see that file or those directories. Place the MP4 in the debug folder and use the path /debug/my_video.mp4

**Reply by tjing11** [reply]
Oh...Yes... you are right. I put the video in the right place but called the host path instead of the container path. I got it to work now. 
So now I have other questions: 
How does the snapshot work? It only is triggered by an object as a person? 
Can I change the object to other stuff? Thanks.

**Reply by NickM-27** [reply]
Update your config with the objects you want to track

```yaml
objects:
 track:
    - person
    - car
    - cat
    - dog
...
```

**Reply by tjing11** [reply]
> Update your config with the objects you want to track

Thanks for the help. it's working now.
